lib/rubocop/cop/performance/inefficient_hash_search.rb

```diff
@@ -42,7 +42,10 @@ class InefficientHashSearch < Cop
         PATTERN
 
         def on_send(node)
-          add_offense(node, message: msg(node)) if inefficient_include?(node)
+          return unless inefficient_include?(node)
+          return if node.children[0].receiver.nil?
```

Comment:
I'd use node destructuring here to highlight what `node.children[0]` stands for.

Reply:
Or node extensions. 😬

Reply:
You can actually make this part of the node pattern:

```ruby
def_node_matcher :inefficient_include?, <<-PATTERN
  (send $(send !nil? ${:keys :values}) :include? $_)
PATTERN
```

In all other places, `children[0]` can be replaces with `receiver`, because we're dealing with `send` nodes throughout.

Reply:
Thanks for review. And I noticed some unnecessary capture, so I removed them.